File: src/examples/java/org/nodex/java/examples/echo/EchoClient.java
/*
 * Copyright 2011 VMware, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.nodex.java.examples.echo;

import org.nodex.java.core.Handler;
import org.nodex.java.core.NodexMain;
import org.nodex.java.core.buffer.Buffer;
import org.nodex.java.core.net.NetClient;
import org.nodex.java.core.net.NetSocket;

public class EchoClient extends NodexMain {

  public static void main(String[] args) throws Exception {
    new EchoClient().run();

    System.out.println("Hit enter to exit");
    System.in.read();
  }

  public void go() throws Exception {
    new NetClient().connect(8080, "localhost", new Handler<NetSocket>() {
      public void handle(NetSocket socket) {

        socket.dataHandler(new Handler<Buffer>() {
          public void handle(Buffer buffer) {
            System.out.println("Net client receiving: " + buffer.toString("UTF-8"));
          }
        });

        //Now send some dataHandler
        for (int i = 0; i < 10; i++) {
          String str = "hello" + i + "\n";
          System.out.print("Net client sending: " + str);
          socket.write(Buffer.create(str));
        }
      }
    });
  }
}

File: src/examples/scala/org/vertx/scala/examples/echo/EchoClient.scala
package org.vertx.scala.examples.echo


/**
 * @author janmachacek
 */

import org.vertx.java.core.Handler
import org.vertx.java.core.buffer.Buffer
import org.vertx.java.core.net.NetSocket
import org.vertx.java.deploy.Verticle

class EchoClient extends Verticle {

  def start() {
    vertx.createNetClient.connect(1234, "localhost", new Handler[NetSocket] {
      def handle(socket: NetSocket) {
        socket.dataHandler(new Handler[Buffer] {
          def handle(buffer: Buffer) {
            System.out.println("Net client receiving: " + buffer)
          }
        })

        for (i <- 0 until 10) {
          val str: String = "hello" + i + "\n"
          System.out.print("Net client sending: " + str)
          socket.write(new Buffer(str))
        }
      }
    })
  }
}

